src/plugins/components/avatar.ts

```diff
@@ -0,0 +1,717 @@
+
+              [`.${prefix}-avatar-dot`]: {
+                [`@apply end-${config.avatar3XL.curvedOrInner.dotPosition} top-${config.avatar3XL.curvedOrInner.dotPosition}`]:
+                  {},
+              },
+            },
+            [`&.${prefix}-avatar-full`]: {
+              [`.${prefix}-avatar-dot`]: {
+                [`@apply end-${config.avatar3XL.fullDotPosition} top-${config.avatar3XL.fullDotPosition}`]:
+                  {},
+              },
+            },
+          },
+          [`&.${prefix}-avatar-4xl`]: {
+            [`@apply h-${config.avatar4XL.size} w-${config.avatar4XL.size}`]:
+              {},
+
+            [`.${prefix}-avatar-badge`]: {
+              [`@apply h-${config.avatar4XL.badge.size} w-${config.avatar4XL.badge.size} -bottom-${config.avatar4XL.badge.position} -end-${config.avatar4XL.badge.position}`]:
+                {},
+            },
+            [`.${prefix}-avatar-dot`]: {
+              [`@apply h-${config.avatar4XL.dot.size} w-${config.avatar4XL.dot.size}`]:
+                {},
+            },
+            [`.${prefix}-avatar-text`]: {
+              [`@apply text-${config.avatar4XL.text}`]: {},
+            },
+            [`&.${prefix}-avatar-rounded, .${prefix}-avatar-inner`]: {
+              [`@apply rounded-${config.avatar4XL.roundedOrInner.rounded}`]: {},
+
+              [`.${prefix}-avatar-dot`]: {
+                [`@apply end-${config.avatar4XL.roundedOrInner.dotPosition} top-${config.avatar4XL.roundedOrInner.dotPosition}`]:
+                  {},
+              },
+            },
+            [`&.${prefix}-avatar-curved, .${prefix}-avatar-inner`]: {
+              [`@apply rounded-${config.avatar4XL.curvedOrInner.rounded}`]: {},
+
+              [`.${prefix}-avatar-dot`]: {
+                [`@apply end-${config.avatar4XL.curvedOrInner.dotPosition} top-${config.avatar4XL.curvedOrInner.dotPosition}`]:
+                  {},
+              },
+            },
+            [`&.${prefix}-avatar-full`]: {
+              [`.${prefix}-avatar-dot`]: {
+                [`@apply end-${config.avatar4XL.fullDotPosition} top-${config.avatar4XL.fullDotPosition}`]:
+                  {},
+              },
+            },
+          },
+
+          [`&.${prefix}-avatar-full`]: {
+            [`@apply rounded-${config.avatarFull.rounded}`]: {},
+
+            [`.${prefix}-avatar-inner`]: {
+              [`@apply rounded-${config.avatarFull.avatarInnerRounded}`]: {},
+            },
+
+            [`.${prefix}-avatar-badge`]: {
+              [`@apply -bottom-${config.avatarFull.avatarBadgePosition} -end-${config.avatarFull.avatarBadgePosition}`]:
+                {},
+            },
+          },
+
+          [`&.${prefix}-avatar-mask:not(.${prefix}-avatar-rounded):not(.${prefix}-avatar-curved):not(.${prefix}-avatar-full)`]:
+            {
+              [`@apply nui-mask`]: {},
+
+              [`&.${prefix}-mask-hex`]: {
```

Comment:
Check the latest source, remove all avatar mask utilities, only leave ${prefix}-avatar-mask

Reply:
Pushed a fix to remove circular mask dependencies